src/Raven.Server/RavenServerStartup.cs

```diff
@@ -218,11 +244,69 @@ private async Task RequestHandler(HttpContext context)
                 }
             }
         }
+        /// <summary>
+        /// IsQuery method checks if watched URL contains "?query" OR "multi_get" OR "index-fields"
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns>boolean</returns>
+        private static bool IsQuery(HttpContext context)
+        {
+            return (
+                    context.Request.QueryString.Value.StartsWith("?query") 
+                    || context.Request.Path.Value.EndsWith("multi_get") 
+                    || context.Request.Path.Value.EndsWith("index-fields")
+                    // add more options
+                    );
+        }
+
+        /// <summary>
+        /// LogTrafficWatch
+        /// </summary>
+        /// <param name="context"></param>
+        /// <param name="elapsedMilliseconds"></param>
+        /// <param name="database"></param>
+        /// <param name="queryOperation"></param>
 
-        private void LogTrafficWatch(HttpContext context, long elapsedMilliseconds, string database)
+        private void LogTrafficWatch(HttpContext context, long elapsedMilliseconds, string database, bool queryOperation)
         {
             var requestId = Interlocked.Increment(ref _requestId);
+            
+            string customInfoString;
+            if (context.Request.Method == "GET" && queryOperation)                          // GET method and URL has query <=> query from Raven.Studio
+            {
+                customInfoString = context.Request.QueryString.Value;
+                customInfoString = Uri.UnescapeDataString(customInfoString.Substring(7, context.Request.QueryString.Value.Length - 65));
+            }
+            else if (context.Request.Method == "POST" && queryOperation)                  // POST method AND URL has query <=> Query from Raven.Client
```

Comment:
We want to do this from the handler, not here. 
The handler gets to decide what goes on in here.

Reply:
done